File: src/main/java/com/prmncr/normativecontrol/components/DocumentParserBuilder.java
package com.prmncr.normativecontrol.components;

import com.prmncr.normativecontrol.services.DocumentParser;
import lombok.AllArgsConstructor;
import org.docx4j.openpackaging.packages.WordprocessingMLPackage;
import org.springframework.stereotype.Component;

@Component
@AllArgsConstructor
public class DocumentParserBuilder {
    private CorrectDocumentParams params;
    private HeadersKeywords keywords;

    public DocumentParser build(WordprocessingMLPackage docx) {
        return new DocumentParser(docx, params, keywords);
    }
}

File: src/main/kotlin/com/prmncr/normativecontrol/components/DocumentParserBuilder.kt
package com.prmncr.normativecontrol.components

import com.prmncr.docx4nc.DocumentParser
import org.docx4j.openpackaging.packages.WordprocessingMLPackage
import org.springframework.stereotype.Component

@Component
class DocumentParserBuilder(private val params: CorrectDocumentParams, private val keywords: SectorKeywords) {

    fun build(docx: WordprocessingMLPackage?): DocumentParser {
        if (docx != null) {
            return DocumentParser(docx, params, keywords)
        } else throw IllegalArgumentException()
    }
}